engine/docker/docker.go

```diff
@@ -93,17 +95,7 @@ func (e *Docker) Setup(ctx context.Context, pipelineConfig *spec.PipelineConfig)
 		}
 	}
 
-	// creates the default pod network. All containers
-	// defined in the pipeline are attached to this network.
-	driver := "bridge"
-	if pipelineConfig.Platform.OS == "windows" {
-		driver = "nat"
-	}
-	_, err := e.client.NetworkCreate(ctx, pipelineConfig.Network.ID, types.NetworkCreate{
-		Driver:  driver,
-		Options: pipelineConfig.Network.Options,
-		Labels:  pipelineConfig.Network.Labels,
-	})
+	err := e.createNetworkWithRetries(ctx, pipelineConfig)
```

Comment:
Do we need this if we already have retries at the top level?

Reply:
we don't have retries at the top level.